src/babel/transformation/file/index.js

```diff
@@ -413,6 +413,30 @@ export default class File {
     return uid;
   }
 
+  addTemplateObject(helperName: string, strings: Array, raw: Array): Object {
+    // Generate a unique name based on the string literals so we dedupe
+    // identical strings used in the program.
+    var stringIds = raw.elements.map(function(string) {
+      return string.value;
+    });
+    var name = `${helperName}_${raw.elements.length}_${stringIds.join(",")}`;
```

Comment:
Yeah, a lookup table is probably the best bet. All that would be required actually would be a different argument given to `generateUidIdentifier` since `this.declarations` is already being used with the complete key.